### PR TITLE
Deck and Other Things

### PR DESCRIPTION
**Pull request by tramstheman**
Created Deck class and documentation. Created README. Minor updates to Card class.

### CODE REVIEW

File: src/com/northvine/engineering/cards/Hand.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    * was successfully added to the hand.
+    */
+   public boolean takeCard(final Card card)
+   {
+      /*
+       Set the card at the top of the array to a deep copy of the card
+       increment the number of cards and then return whether or not it was
+       successful by calling the negation of the deep copy's error flag.
+        */
+      return !(myCards[numCards++] = new Card(card)).getErrorFlag();
```
Comment (KyleLuoma) [inline src/com/northvine/engineering/cards/Hand.java]: takeCard() allows an out of bound condition if the method is called more times than MAX_CARDS. I recommend adding a condition check to ensure numCards < MAX_CARDS before attempting to append another card to the end of myCards[].

File: src/com/northvine/engineering/cards/Hand.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    * This method returns the top card of the hand. In other words,
+    * the highest index of the deck is returned. This card is also
+    * removed from the list of cards.
+    *
+    * @return the "top" card in the hand
+    */
+   public Card playCard()
+   {
+      Card card = new Card(myCards[--numCards]); // cache temp location of top card deep copy
+      myCards[numCards] = null; // implicitly call destructor by re-assigning to null
```
Comment (KyleLuoma) [inline src/com/northvine/engineering/cards/Hand.java]: Allows an index out of bound exception if playCard() is called on an empty hand. I guess exception handling would take care of this, so I am not sure if we want to keep it as-is or not. Just figured I'd point it out.

File: src/com/northvine/engineering/cards/Hand.java (hunk 1 above)
Comment (tramstheman) [inline src/com/northvine/engineering/cards/Hand.java]: Yeah, I saw in the forum that this was a requirement, although it is not explicitly declared in the specs page. I will make this fix.

File: src/com/northvine/engineering/cards/Hand.java (hunk 2 above)
Comment (tramstheman) [inline src/com/northvine/engineering/cards/Hand.java]: The project requirements did not explicitly declare this as a requirement. I wasn't sure if he actually wanted it or not, but I will push with a check in as well.

**Pull request merged**